lib/utils/report.mjs

```diff
@@ -44,37 +46,23 @@ export default function report(problem) {
 		);
 	}
 
-	const { ignoreDisables } = result.stylelint.config || {};
 	const { disabledRanges } = result.stylelint;
 
-	if (disabledRanges) {
-		const ranges = disabledRanges[ruleName] ?? disabledRanges[RULE_NAME_ALL] ?? [];
-
-		for (const range of ranges) {
-			if (
-				// If the problem is within a disabledRange,
-				// and that disabledRange's rules include this one,
-				// do not register a warning
-				range.start <= startLine &&
-				(range.end === undefined || range.end >= startLine) &&
-				(!range.rules || range.rules.includes(ruleName))
-			) {
-				// Collect disabled warnings
-				// Used to report `needlessDisables` in subsequent processing.
-				const disabledWarnings =
-					result.stylelint.disabledWarnings || (result.stylelint.disabledWarnings = []);
-
-				disabledWarnings.push({
-					rule: ruleName,
-					line: startLine,
-				});
-
-				if (!ignoreDisables) {
-					return;
-				}
-
-				break;
-			}
+	if (isDisabled(ruleName, startLine, disabledRanges)) {
+		// Collect disabled warnings
+		// Used to report `needlessDisables` in subsequent processing.
+		const disabledWarnings =
+			result.stylelint.disabledWarnings || (result.stylelint.disabledWarnings = []);
```

Comment:
[note] It seems reasonable that the initialization of `disabledWarnings` is performed like other `PostcssResult.stylelint` properties (see #7728). After this PR is merged, let's consider if there is any chance.

Reply:
Yes please, postpone it.